.contrib/Source Code/Parser/Framework/Framework.Processing.cs

```diff
@@ -3879,208 +3878,107 @@ private static void IncrementTypeUseCount(string key, decimal id)
-                                    }
+            // We don't want things that got deleted to be in the addon.
+            // NOTE: If it's not the last entry, that means it might have been readded later?
+            // CRIEVE NOTE: Braghe wanted Debug Mode to not completely delete a thing from the exported Debug files...
+            // Deleting it from the actual database is actually expected for the real builds,
+            // so don't remove this. This is how I want it. Thanks!
+            if (adaptedTimeline.RemovedStatus == RemovedStatus.DELETED_FROM_GAME && !DebugMode)
+                return false;   // Invalid entry
 
-                                    // just in case parsing with Debug and verifying in-game... we would want Deleted to show as Removed
-                                    removed = 4;
-                                    readded = false;
-                                    // Mark the first patch this was removed on. (the upcoming patch)
-                                    if (removedPatch <= 10000)
-                                    {
-                                        timeline.CurrentEntry = index;
-                                        removedPatch = entry.Version;
-                                    }
-                                }
-                                break;
-                            }
-                        case "removed":
-                            {
-                                if (CURRENT_RELEASE_VERSION >= entry.LongVersion)
-                                {
-                                    removed = 2;
-                                    readded = false;
-                                    // Mark the most recent patch this was removed
-                                    if (removedPatch <= 10000)
-                                    {
-                                        timeline.CurrentEntry = index;
-                                        removedPatch = entry.Version;
-                                    }
-                                }
-                                else
-                                {
-                                    // Mark the first patch this was removed on. (the upcoming patch)
-                                    if (removedPatch <= 10000) removedPatch = entry.Version;
-                                }
-                                break;
-                            }
-                    }
-                }
+            // Undo NYI status if we're processing an unsorted category
+            if (adaptedTimeline.RemovedStatus == RemovedStatus.NEVER_IMPLEMENTED && ProcessingUnsortedCategory)
+                adaptedTimeline.RemovedStatus = 0;
 
-                // final removed type for the current parser patch
-                switch (removed)
+            // Set current entry for timeline
+            timeline.SetCurrentEntry(adaptedTimeline.Entries[0]);
```

Comment:
Do we know that the first entry of the `AdaptedTimeline` will always be the most recent added/removed patch of the Thing? This property is only used in one place currently (to determine `coords` shifting based on map changes from patch to patch) but the 'CurrentEntry' is supposed to represent the patch which 'matches' the current status of that Thing in the game.

The logic for the `AdaptedTimeline` is pretty complex, but I guess because it's extracting the previous looping timeline logic from `CheckTimeline` so it's a bit tricky to follow at first glance.